### ISSUE TITLE
win32: bug in date conversion due to time_t and gmtime not working with dates before Jan 1st, 1970

### ISSUE THREAD

**Issue by xdeng** [libimobiledevice/libplist#56]
demo file http://xdeng.cn/file/time2.plist

mac parse  1931-02-12

libplist  Can't parse

**Comment by nikias** [comment]
@xdeng have you tested with latest git HEAD? It works correctly for me.

**Comment by xdeng** [comment]
@nikias The latest version also can not be resolved        You try demo file http://xdeng.cn/file/time2.plist

**Comment by nikias** [comment]
This is the output I get:
```
$ plistutil -i time2.plist 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<array>
	<string>SDMessageProcessChanges</string>
	<string>com.apple.Contacts</string>
	<dict>
		<key>1</key>
		<dict>
			<key>last name</key>
			<string>小</string>
			<key>birthday</key>
			<date>1931-02-12T20:00:00Z</date>
			<key>com.apple.syncservices.RecordEntityName</key>
			<string>com.apple.contacts.Contact</string>
			<key>first name</key>
			<string>邓</string>
			<key>display as company</key>
			<string>person</string>
		</dict>
	</dict>
	<true/>
	<string>___EmptyParameterString___</string>
</array>
</plist>
```
Looks correct to me.

**Comment by xdeng** [comment]
@nikias  Your machine is 64 bit ????

**Comment by xdeng** [comment]
plist_t plist_new_date(int32_t sec, int32_t usec);   change plist_t plist_new_date(int64_t sec, int64_t usec); ??????????

**Comment by xdeng** [comment]
in xplist.c 232 line

The Windows version

struct tm *gmtime( 
   const time_t *timer 
);
If timer represents a date before midnight, January 1, 1970, gmtime returns NULL.

so 1931 year can't parse

**Comment by nikias** [comment]
I see the problem. Not sure yet how we should solve this across platforms though.

**Comment by nikias** [comment]
Actually there is also a problem on macOS, mktime (used in xplist.c to parse dates from XML) does not work with dates before ~1900, even though the time_t is 64 bit signed and would easily support that.
I think I will also change the internal storage from struct timeval to double - this is what the binary plist format uses - to get rid of the 32 vs 64 bit storage size difference (and the actual 8 bytes overhead for the plist_data structure in case of 2x64 bit struct timeval size).
Also I found this: https://github.com/evalEmpire/y2038 which would make it work on 32 bit platforms, I tested this on windows and 32-bit linux and it works perfectly.
